packages/server/tests/MpeGetContext.test.ts

```diff
@@ -0,0 +1,412 @@
+import Database from '@ioc:Adonis/Lucid/Database';
+import MpeServerToTemporalController from 'App/Controllers/Http/Temporal/MpeServerToTemporalController';
+import { datatype, random } from 'faker';
+import test from 'japa';
+import sinon from 'sinon';
+import {
+    initTestUtils,
+    generateMpeWorkflowState,
+    createSpyOnClientSocketEvent,
+} from './utils/TestUtils';
+
+test.group('MPE get context', (group) => {
+    const {
+        createUserAndGetSocket,
+        disconnectEveryRemainingSocketConnection,
+        initSocketConnection,
+        waitFor,
+    } = initTestUtils();
+
+    group.beforeEach(async () => {
+        initSocketConnection();
+        await Database.beginGlobalTransaction();
+    });
+
+    group.afterEach(async () => {
+        await disconnectEveryRemainingSocketConnection();
+        sinon.restore();
+        await Database.rollbackGlobalTransaction();
+    });
+
+    test('It should retrieve context from an already joined room', async (assert) => {
+        const creatorUserID = datatype.uuid();
+        const roomsIDs = Array.from({
```

Comment:
There is a `generateArray` function in `TestUtils.ts` that we can use to create an array of random size.